pkg/repositories/transformers/node_execution.go

```diff
@@ -317,3 +324,29 @@ func GetNodeExecutionInternalData(internalData []byte) (*genModel.NodeExecutionI
 	}
 	return &nodeExecutionInternalData, nil
 }
+
+func handleNodeExecutionInputs(ctx context.Context,
+	nodeExecutionModel *models.NodeExecution,
+	request *admin.NodeExecutionEventRequest,
+	storageClient *storage.DataStore) error {
+	if len(nodeExecutionModel.InputURI) > 0 {
+		// Inputs are static over the duration of the node execution, no need to update them when they're already set
+		return nil
+	}
+	if request.Event.GetInputData() != nil {
```

Comment:
Does it make sense to use something like `switch input := request.Event.GetInputValue().(type)` here instead of checking on `nils` to reinforce the `oneof` in the proto definition?

Reply:
sure, done